homeassistant/components/smarty.py

```diff
@@ -0,0 +1,220 @@
+        'MULTIPLIER': 1,
+        'STATE': {
+            0: 'Stand-by',
+            1: 'Building protection',
+            2: 'Economy',
+            3: 'Comfort',
+            4: 'Boost'
+        },
+        'UNIT_OF_MESUREMENT': None
+    },
+    'IR_SUPPLY_AIR_TEMPERATURE': {
+        'ADDR': 18,
+        'NAME': 'T1-Supply air temperature',
+        'MULTIPLIER': 0.1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': TEMP_CELSIUS
+    },
+    'IR_EXTRACT_AIR_TEMPERATURE': {
+        'ADDR': 19,
+        'NAME': 'T2-Extract air temperature',
+        'MULTIPLIER': 0.1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': TEMP_CELSIUS
+    },
+    'IR_OUTDOOR_AIR_TEMPERATURE': {
+        'ADDR': 21,
+        'NAME': 'Outdoor air temperature',
+        'MULTIPLIER': 0.1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': TEMP_CELSIUS
+    },
+    'IR_SUPPLY_FAN_SPEED_RPM': {
+        'ADDR': 55,
+        'NAME': 'Supply fan spreed RPM',
+        'MULTIPLIER': 1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': RPM
+    },
+    'IR_EXTRACT_FAN_SPEED_RPM': {
+        'ADDR': 56,
+        'NAME': 'Extract fan spreed RPM',
+        'MULTIPLIER': 1,
+        'STATE': {},
+        'UNIT_OF_MESUREMENT': RPM
+    }
+}
+
+HOLDING_REGS = {
+    'HR_USER_CONFIG_CURRENT_SYSTEM_MODE': {
+        'ADDR': 1,
+        'NAME': 'Current system mode',
+        'STATE': {
+            'Stand-by': 0,
+            'Building protection': 1,
+            'Economy': 2,
+            'Confort': 3
+        }
+    }
+}
+
+
+def setup(hass, config):
+    """Set up the smarty environment."""
+    conf = config[DOMAIN]
+
+    name = conf.get(CONF_NAME)
+    modbus_slave = conf.get(CONF_SLAVE)
+
+    _LOGGER.debug("name: %s, modbus_slave: %s", name, modbus_slave)
+
+    smarty = Smarty(hass, name, modbus_slave)
+    hass.data[DATA_SMARTY] = smarty
+
+    # Load platforms
+    discovery.load_platform(hass, 'fan', DOMAIN, {}, config)
+    discovery.load_platform(hass, 'sensor', DOMAIN, {}, config)
+
+    def poll_device_update(event_time):
+        """Update Smarty device."""
+        _LOGGER.debug("Updating Smarty device...")
+        smarty.update()
+        dispatcher_send(hass, SIGNAL_UPDATE_SMARTY)
+
+    track_time_interval(hass, poll_device_update, timedelta(seconds=30))
+
+    return True
+
+
+class Smarty:
+    """Representation of a Smarty Ventilation Unit."""
+
+    def __init__(self, hass, name, modbus_slave):
+        """Initialize the Smarty Ventilation Unit."""
+        self.data = {}
+        self.name = name
+        self.hass = hass
+        self._modbus_slave = modbus_slave
+        self._holding_registers = []
+        self._coils = []
+        self._discrete_inputs = []
+        self._input_registers = []
+
+    def update(self):
+        """Update registers."""
+        from pymodbus.exceptions import ConnectionException
+        try:
+            result = modbus.HUB.read_holding_registers(unit=self._modbus_slave,
```

Comment:
This is way too much specific protocol specific information for Home Assistant. All the modbus specific constants and these calls should be extracted into a PyPi package. You should have the PyPi package export a class that allows passing in a modbus hub.

```python
from pysmarty import Smarty

data = Smarty(modbus.HUB)
data.update()
```

Reply:
ok, I'll do that. I did it this way as it depends on the homeassistant modbus component. My understanding was that a pypi package would be working also as stand-alone.